### PR TITLE
EZP-26191: Draft conflict screen on dashboard

### PR DESCRIPTION
**Pull request by yannickroger**
Link: https://jira.ez.no/browse/EZP-26191

## Description
Adds the draft conflict screen to the edit content buttons of the dashboard.

Screencast: https://drive.google.com/file/d/0B5amepf6EkXbQzhKME1EZkxHUmc/view

## Tests
Manual and unit tests.

### CODE REVIEW

**Comment by yannickroger** [comment]
I am having a vision of @dpobel drinking his cofee tomorrow morning having a look at this PR :smile:

File: Resources/config/yui.yml [hunk 1]
```diff
@@ -333,13 +333,20 @@ system:
                     requires: ['ez-dashboardblockbaseview', 'ez-asynchronousview', 'event-tap', 'event-outside']
                     path: "%ez_platformui.public_dir%/js/views/dashboard/ez-dashboardblockasynchronousview.js"
                 ez-dashboardblockallcontentview:
-                    requires: ['ez-dashboardblockasynchronousview', 'dashboardblockallcontentview-ez-template']
+                    requires:
+                        - 'ez-dashboardblockasynchronousview'
+                        - 'dashboardblockallcontentview-ez-template'
+                        - 'ez-draftconflictbutton'
```
Comment (dpobel) [inline Resources/config/yui.yml]: it's not a matter of *button* anymore, I would name the module `ez-draftconflict` and the corresponding extension `Y.eZ.DraftConflict`

File: Resources/public/js/extensions/ez-draftconflictbutton.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+
+    /**
+     * Views extension providing the draft conflict feature to an edit button
+     *
+     * When a view is extended with this extension it needs to add an event on the button that
+     * calls `_fireEditContentRequest` when fired.
+     * Also, the `_getContentEditRequestParams` method need to be implemented in the view.
+     *
+     * container scales to match the viewport's height new dimension.
```
Comment (dpobel) [inline Resources/public/js/extensions/ez-draftconflictbutton.js]: ?

File: Resources/public/js/extensions/ez-draftconflictbutton.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+     * @class HeightFit
+     * @extensionfor Y.View
+     */
+    Y.eZ.DraftConflictButton = Y.Base.create('draftConflictButtonExtension', Y.View, [], {
+
+        /**
+         * Provides the parameters needed to the content edit request
+         *
+         * @method _getContentEditRequestParams
+         * @param {Y.Node} target the target that has been clicked
```
Comment (dpobel) [inline Resources/public/js/extensions/ez-draftconflictbutton.js]: it's not necessarily a click so I would document that as: *target element of the event triggering the edit request*

File: Resources/public/js/extensions/ez-draftconflictbutton.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    /**
+     * The draft conflict button extension
+     *
+     * @module ez-draftconflictbutton
+     */
+    Y.namespace('eZ');
+
+
+    /**
+     * Views extension providing the draft conflict feature to an edit button
```
Comment (dpobel) [inline Resources/public/js/extensions/ez-draftconflictbutton.js]: `Views extension providing the draft conflict feature.`

File: Resources/public/js/extensions/ez-draftconflictbutton.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+            e.preventDefault();
+
+            params = this._getContentEditRequestParams(e.target);
+
+            /**
+             * Fired when a content needs to be edited
+             * @event editContentRequest
+             */
+            this.fire('editContentRequest', {
+                content: params.contentInfo,
```
Comment (dpobel) [inline Resources/public/js/extensions/ez-draftconflictbutton.js]: a `content` that references a Content Info, that's not super clean

File: Resources/public/js/views/services/plugins/ez-contenteditplugin.js [hunk 6]
```diff
@@ -34,7 +34,7 @@ YUI.add('ez-contenteditplugin', function (Y) {
          * @method _checkExistingDraft
          * @protected
          * @param {Object} e event facade of the editContentRequest event
-         * @param {eZ.Content} e.content being edited
+         * @param {eZ.Content|eZ.ContentInfo} e.content being edited
```
Comment (dpobel) [inline Resources/public/js/views/services/plugins/ez-contenteditplugin.js]: it's the counter part of https://github.com/ezsystems/PlatformUIBundle/pull/673/files#r78139237 this method should be able to accept either the Content or Content Info but in 2 different property of the event facade.

**Comment by yannickroger** [comment]
Ready for review @dpobel and @StephaneDiot 
I will do the screencast in a couple of hours.

File: Resources/config/yui.yml [hunk 7]
```diff
@@ -1322,12 +1331,18 @@ system:
                 ez-translateproperty:
                     requires: ['base', 'array-extras']
                     path: "%ez_platformui.public_dir%/js/extensions/ez-translateproperty.js"
-                ez-contentinfo-attributes:
-                    requires: ['ez-restmodel']
+                ez-contentinfo-attributes: # deprecated module, provide a BC component based on ez-contentinfo-base
+                    requires: ['ez-restmodel', ez-contentinfo-base]
```
Comment (dpobel) [inline Resources/config/yui.yml]: `ez-restmodel` is not needed anymore

File: Resources/public/js/views/confirmbox/ez-draftconflictview.js [hunk 8]
```diff
@@ -42,11 +42,24 @@ YUI.add('ez-draftconflictview', function (Y) {
...
+                contentInfo = this.get('contentInfo'),
+                contentJson,
+                contentInfoJson;
+
+            if (contentInfo) {
+                contentInfoJson = contentInfo.toJSON();
+            }
+
+            if (content.toJSON) {
+                contentJson = content.toJSON();
```
Comment (dpobel) [inline Resources/public/js/views/confirmbox/ez-draftconflictview.js]: you can add the deprecated warning here

File: Resources/public/js/views/dashboard/ez-dashboardblockallcontentview.js [hunk 9]
```diff
@@ -22,9 +27,12 @@ YUI.add('ez-dashboardblockallcontentview', function (Y) {
      * @constructor
      * @extends eZ.DashboardBlockAsynchronousView
      */
-    Y.eZ.DashboardBlockAllContentView = Y.Base.create('dashboardBlockAllContentView', Y.eZ.DashboardBlockAsynchronousView, [], {
+    Y.eZ.DashboardBlockAllContentView = Y.Base.create(
+        'dashboardBlockAllContentView', Y.eZ.DashboardBlockAsynchronousView, [Y.eZ.DraftConflict], {
```
Comment (dpobel) [inline Resources/public/js/views/dashboard/ez-dashboardblockallcontentview.js]: weird indentation ?

File: Resources/public/js/views/dashboard/ez-dashboardblockmycontentview.js [hunk 10]
```diff
@@ -35,6 +41,19 @@ YUI.add('ez-dashboardblockmycontentview', function (Y) {
             };
         },
 
+        _callFireEditContentRequest: function (e) {
```
Comment (dpobel) [inline Resources/public/js/views/dashboard/ez-dashboardblockmycontentview.js]: missing JS doc

File: Tests/js/views/dashboard/assets/ez-dashboardblockasynchronousview-tests.js [hunk 11]
```diff
@@ -129,6 +129,120 @@ YUI.add('ez-dashboardblockasynchronousview-tests', function (Y) {
             this.wait();
         },
     };
+
+    NS.editContentEventTest = {
```
Comment (dpobel) [inline Tests/js/views/dashboard/assets/ez-dashboardblockasynchronousview-tests.js]: `NS.EditContentEventTest`

**Comment by yannickroger** [comment]
@dpobel 's feedback taken into account

**Pull request merged**